contracts/InsuranceArbitrator.sol

```diff
@@ -22,16 +29,22 @@ contract InsuranceArbitrator {
         bool claimInitiated; // Claim state preventing simultaneous claim attempts.
         string insuredEvent; // Short description of insured event.
         address insuredAddress; // Beneficiary address eligible for insurance compensation.
-        address currency; // Denomination token for insurance coverage.
+        IERC20 currency; // Denomination token for insurance coverage.
         uint256 insuredAmount; // Amount of insurance coverage.
     }
 
+    // Tracks raised claims on insurance policies.
+    struct Claim {
+        bytes32 policyId; // Claimed policy identifier.
+        OptimisticOracleV2Interface optimisticOracle; // optimistic oracle instance where claims are resolved.
```

Comment:
Its required to store OO implementation per claim in order to handle situation where OO is upgraded in the Finder by governance. This contract would still use old OO for claims initiated against the old OO. Though it assumes old OO is not removed from the registry and still can handle DVM requests.